include/openthread/link_metrics.h

```diff
@@ -92,11 +81,23 @@ typedef struct otLinkMetricsSeriesFlags
     bool mMacAck : 1;         ///< MAC Ack.
 } otLinkMetricsSeriesFlags;
 
+/**
+ * Enhanced-ACK Flags.
+ *
+ * These are used in Enhanced-ACK Based Probing to indicate whether to register or clear the probing.
+ *
+ */
+typedef enum otLinkMetricsEnhAckFlags
+{
+    OT_LINK_METRICS_ENH_ACK_CLEAR    = 0, ///< Clear.
+    OT_LINK_METRICS_ENH_ACK_REGISTER = 1, ///< Register.
+} otLinkMetricsEnhAckFlags;
+
 /**
  * Link Metrics Status values.
  *
  */
-typedef enum otLinkMetricsStatus : uint8_t
+typedef enum otLinkMetricsStatus
```

Comment:
what's the reason for this change?

Just noticing that `aStatus` is of type `uint8_t` in below two callback declarations, better to update to be of type `otLinkMetricsStatus` (fine to me if you prefer a separate PR for this).

https://github.com/openthread/openthread/blob/6fda0786edab8e821e1647cc4be08b4a7059cf4a/include/openthread/link_metrics.h#L118-L130

Reply:
The reason is that C compiler doesn't support this. The platform code including this header are compiled with C compiler. If we add it, we got such error:
```
In file included from ../../../examples/platforms/utils/link_metrics.h:42:0,
                 from ../../../examples/platforms/simulation/radio.c:43:
../../../include/openthread/link_metrics.h:100:14: error: ISO C forbids forward references to 'enum' types [-Wpedantic]
 typedef enum otLinkMetricsStatus : uint8_t
              ^~~~~~~~~~~~~~~~~~~
../../../include/openthread/link_metrics.h:100:34: error: expected identifier or '(' before ':' token
 typedef enum otLinkMetricsStatus : uint8_t
                                  ^
../../../include/openthread/link_metrics.h:108:3: error: data definition has no type or storage class
 } otLinkMetricsStatus;
```

This error is not found in last PR because at that time the radio doesn't use the link_metrics module which includes the public link metrics header.

Reply:
Yes. (If I recall correctly) specifying `enum` size is supported from C++11 onwards.